### DISCUSSION TITLE
Advice on Zero W 2 cooling and powering

### DISCUSSION THREAD

**Discussion by jonnywright** [marcone/teslausb#857] [Q&A] [answered]
Hello friends,

I only very recently stumbled upon this project and I am quite excited to implement it. I have a couple of questions which I hope the community can answer. I am in a 2023 Model Y, UK spec.

1. I intend to use a Pi Zero 2 W. Am I likely to require any kind of cooling? I am looking at cases and am wondering whether I need some kind of passive cooling case, or if I'd be OK with a basic 3D printed case.
2. Reading the [Powering TeslaUSB](https://github.com/marcone/teslausb/wiki/Power) page, it briefly references that if powering directly from the USB port (which I plan to do, in the glovebox USB port), you can ensure that it stays powered by "_turning on climate control or Sentry mode_". I keep Sentry mode enabled 24/7, and just want to confirm that by virtue of having Sentry mode enabled, my unit would be constantly powered?

Thanks a lot for reading. I know these may seem obvious but I just want confirmation before committing to hardware, and hopefully these haven't been asked before - I did do a search first!

**Comment by theendofsilence** [comment, marked as answer]
Hi Jonny,

1. I'm in the exact same position - have just bought and configured my Pi Zero 2 - I've used a [metal case](https://www.amazon.co.uk/dp/B0CHB5MH8C?psc=1&ref=ppx_yo2ov_dt_b_product_details) (£10) - and it seems to keep the device in the operating temperature of 41 degs with no issue.
2. I'd been looking for that link on the power (thanks!)- most of the time Sentry is active, but even then it sometimes appears to be lost when I get in the car the following day. I'm using the glovebox port - and thinking of putting a usb hub it to see if that improves it.

Its cost me £28 all up so far, so definitely worth playing with - there are other projects using the Pi Zero 2 W to get stats out the CAN bus ports too!.

**Reply by jonnywright** [reply]
So I bit the bullet and went with the setup above. I have had the Pi plugged in overnight and it has stayed powered on all night 👍🏽
Also, the temp seems to be sitting at a steady 50c (with no case at all), which is good. I will keep an eye on this, and if the weather gets any hotter I may reconsider the need for cooling.

**Comment by tribhuvanji** [comment]
I also use a Zero 2 W with a metal case for last couple of years. It can get very hot where I live with summer days with ambient outside temp hitting 112F / 44C. That is only magnified on the inside of the car - you could fry an egg in there and a passively cooled Pi would fare about the same. 

On the hottest days I use cabin overheat protection which besides not being foolproof will turn the Pi on while the car is roasting hot. One of my Zero's went offline during that so I decided it's better to just pull the plug when I park and it's over 100F/38C outside. 

I suspect even active cooling will not be enough when the car is that hot. In these temps I use a split config with a standard USB for Sentry and just my music on the teslausb. This way yanking the Pi does not interrupt Sentry.

Fortunately we only get these oven-hot days around one month of the year.
